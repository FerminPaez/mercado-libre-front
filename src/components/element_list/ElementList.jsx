import { useState, useEffect, useRef } from 'react'
import { Link } from 'react-router-dom'
import { getList } from '../../services/services'
import { Element } from '../element/Element'

export const ElementList = ({ search }) => {
  const [elementList, setElementList] = useState('')
  const { current: myArray } = useRef(elementList)
  const [error, setError] = useState('')

  useEffect(() => {
    const setList = async () => {
      if (search !== null && search !== '' && search !== undefined) {
        const respjson = await getList(search)
        console.log(respjson)
        if (respjson === 404) {
          setError(respjson)
        } else setElementList(respjson.items)
      }
    }
    setList()
  }, [search, myArray])

  return (
    <div className='container p-0'>
      {
      elementList !== '' || elementList === undefined || typeof elementList !== 'string'
        ? elementList.map(elemento => {
          return (
            <Link to={`/items/${elemento.id}`} key={elemento.id}>
              <Element props={elemento}/>
            </Link>)
        })
        : error !== 404 ? <p>Cargando</p> : <p>Estamos teniendo problemas para realizar esta busqueda</p>
      }
    </div>
  )
}
