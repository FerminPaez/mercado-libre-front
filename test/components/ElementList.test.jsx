import { render, screen } from '@testing-library/react'
import { ElementList } from '../../src/components/element_list/ElementList'

describe('Suite de pruebas para element', () => {
  test('Should be match with snapshot', () => {
    const { container } = render(<ElementList props={ 'iphone' } />)
    expect(container).toMatchSnapshot()
  })

  test('Should be match with snapshot when the elements are load', async () => {
    render(<ElementList search={ 'iphone' } />)
    expect(screen).toMatchSnapshot()
  })
})
