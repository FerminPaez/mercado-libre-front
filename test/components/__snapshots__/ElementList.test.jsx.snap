// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suite de pruebas para element Should be match with snapshot 1`] = `
<div>
  <div
    class="container p-0"
  >
    <p>
      Cargando
    </p>
  </div>
</div>
`;

exports[`Suite de pruebas para element Should be match with snapshot when the elements are load 1`] = `undefined`;
